Source/WarpX.cpp

```diff
@@ -340,6 +340,24 @@ WarpX::ReadParameters ()
     {
         ParmParse pp("warpx");
 
+        // set random seed
+        std::string random_seed = "default";
+        pp.query("random_seed", random_seed);
+        std::random_device rd;
+        std::uniform_int_distribution<int> dist(0, INT_MAX);
```

Comment:
Can these declarations be moved inside of the if block where they are used? Also, should this be dist(1, INT_MAX), since dist is inclusive and can return the bounding values?